verilog/formatting/token_annotator.cc

```diff
@@ -93,10 +93,9 @@ static WithReason<int> SpacesRequiredBetween(const PreFormatToken& left,
   }
 
   // Unary operators (context-sensitive)
-  if (IsUnaryPrefixExpressionOperand(left, context)) {
-    // TODO(b/143739545): Don't accidentally join into valid tokens, e.g.:
-    //   '-' and '-' could form '--' (decrement).
-    //   '-' and '=' could form '-=' (subtract-assign).
+  if (IsUnaryPrefixExpressionOperand(left, context) &&
+      (left.format_token_enum != FormatTokenType::binary_operator ||
+       !IsUnaryOperator(static_cast<yytokentype>(right.TokenEnum())))) {
```

Comment:
Safe for now, but leave a TODO to allow *some* unary operators on the right to be formatted with 0-space.

Reply:
I've added a TODO, along with an example that `a = & ~b;` could be formatted as `a = &~a;`. I hope that's the kind of situation you had in mind.

BTW There is also an issue that the opposite to my example (`a = ~ &b`) shouldn't really be formatted as 0-space, as `~&` is a separate operator (although I think the results of `~` and `&` would technically be the same as of `~&`, so I'm not sure if we should be bothered about this).

Reply:
Thanks for the TODO test cases.  Conservatively, even if the joined operator is the same, the formatter currently runs a check for lexical equivalence on the formatted output -- it would be easier to verify token-for-token if they remained lexed separately.